npm/src/clientfilemetadataupload/index.ts

```diff
@@ -28,104 +30,101 @@ export class ClientFileMetadataUpload {
     this.client = client
   }
 
-  async saveFileInformation(
-    numberOfFiles: number,
-    uploadFilesInfo: FileUploadInfo
-  ): Promise<string[]> {
-    const variables: AddFilesMutationVariables = {
-      addFilesInput: {
-        consignmentId: uploadFilesInfo.consignmentId,
-        numberOfFiles: numberOfFiles,
-        parentFolder: uploadFilesInfo.parentFolder
-      }
+  async startUpload(uploadFilesInfo: FileUploadInfo): Promise<void> {
+    const variables: StartUploadMutationVariables = {
+      input: uploadFilesInfo
     }
 
-    const result: FetchResult<AddFilesMutation> = await this.client.mutation(
-      AddFiles,
+    const result: FetchResult<StartUploadMutation> = await this.client.mutation(
+      StartUpload,
       variables
     )
 
     if (!result.data || result.errors) {
       const errorMessage: string = result.errors
         ? result.errors.toString()
         : "no data"
-      throw Error(`Add files failed: ${errorMessage}`)
-    } else {
-      return result.data.addFiles.fileIds
+      throw Error(`Start upload failed: ${errorMessage}`)
     }
   }
 
   async saveClientFileMetadata(
-    fileIds: string[],
+    consignmentId: string,
     metadata: IFileMetadata[]
   ): Promise<ITdrFile[]> {
-    const metadataInputs: AddClientFileMetadataInput[] = []
-    const files: ITdrFile[] = []
-    this.generateClientFileData(fileIds, metadata).forEach((value) => {
-      metadataInputs.push(value.metadataInput)
-      files.push(value.tdrFile)
-    })
-
-    const metadataBatches: AddClientFileMetadataInput[][] =
-      this.createMetadataInputBatches(metadataInputs)
-
-    for (const metadataInputs of metadataBatches) {
-      const variables = { input: metadataInputs }
-      const result = await this.client.mutation(
-        AddClientFileMetadata,
-        variables
-      )
-
-      if (result.errors) {
-        throw Error(
-          `Add client file metadata failed: ${result.errors.toString()}`
-        )
+    let matchId = 0
+    const clientFileData: IClientFileData[] = metadata.map((m) => {
```

Comment:
Does mapping with the index work here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#parameters
 
```
const clientFileData: IClientFileData[] =  metadata.map((m, i) => {
  const matchId = i
  ...
})
```

It would save you from having to mutate `matchId` inside the the functional-looking `map`.

Reply:
Yeah you're right, that's better, I've changed it.